### PR TITLE
Rest model transform

### PR DESCRIPTION
**Pull request by Varagos**
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes an bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Bitloops Language?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/bitloops/bitloops-language/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the Bitloops Language CLA (if not, email us: contributor@bitloops.com)
- [ ] I created tests which fail with the change
- [ ] Code complies correctly
- [ ] Unit tests pass locally with my changes
- [ ] All existing tests passed
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added/extended necessary documentation
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: transpiler/src/target-ast/node-transformers/controllers/index.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+class BaseControllerNodeTSTransformer<
+  T extends ControllerNode,
+> extends NodeModelToTargetASTTransformer<T> {
+  run(): void {
+    this.transformAwait();
+    this.transformDotValue();
+  }
+
+  private transformAwait(): void {
+    const executeStatement = this.tree.getUseCaseExecuteStatementOf(this.node);
```
Comment (Varagos) [inline transpiler/src/target-ast/node-transformers/controllers/index.ts:15]: handle variable & just method call

File: transpiler/src/target-ast/node-transformers/controllers/index.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    if (!executeStatement) {
+      return;
+    }
+
+    const expression = executeStatement.getExpression();
+    if (!expression.isMethodCallExpression()) {
+      return;
```
Comment (Varagos) [inline transpiler/src/target-ast/node-transformers/controllers/index.ts:22]: throw error

File: transpiler/src/ast/core/intermediate-ast/nodes/Expression/MethodCallExpression.ts [hunk 3]
```diff
@@ -10,15 +10,42 @@ export class MethodCallExpressionNode extends ExpressionNode {
...
+
+    if (!expression.isMemberDotExpression()) {
+      throw new Error('Expression is not a member dot expression');
+    }
+    // Find leftmost expression of dotMemberExpression
+    const leftMostExpression = expression.getLeftMostExpression();
+    if (!leftMostExpression.isThisExpression()) {
+      throw new Error('Leftmost expression is not a this expression');
+    }
+    leftMostExpression.updateValue('await this');
```
Comment (Varagos) [inline transpiler/src/ast/core/intermediate-ast/nodes/Expression/MethodCallExpression.ts:49]: break get leftmost this, and update of its value

File: transpiler/src/target-ast/node-transformers/controllers/index.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+    expression.prependAwaitToThisMethod();
+    return this.tree.buildValueRecursiveBottomUp(this.node);
+  }
+
+  private transformDotValue(): void {
+    const executeStatement = this.tree.getUseCaseExecuteStatementOf(this.node);
+    if (!executeStatement) {
```
Comment (Varagos) [inline transpiler/src/target-ast/node-transformers/controllers/index.ts:30]: check for 3 cases

File: transpiler/src/target-ast/node-transformers/controllers/index.ts [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      return;
+    }
+
+    const identifierNode = executeStatement.getIdentifier();
+    const identifierValue = identifierNode.getIdentifierName();
+    const newValue = `${identifierValue}.value`;
```
Comment (giorgosnty) [inline transpiler/src/target-ast/node-transformers/controllers/index.ts:36]: newValue could be renamed to something indicating more what this is and the right part maybe a private method if we are going to use it more. If not that's ok.

**Review by giorgosnty — APPROVED** [review]

**Pull request merged**